pkg/apiserver/apic.go

```diff
@@ -283,37 +301,30 @@ func (a *apic) PullTop() error {
 		case "range":
 			*decision.Scope = types.Range
 		}
+		capiPullTopX.Decisions = append(capiPullTopX.Decisions, decision)
+	}
 
-		/*if the scope is IP or Range, convert the value to integers */
-		if strings.ToLower(*decision.Scope) == "ip" || strings.ToLower(*decision.Scope) == "range" {
-			sz, start_ip, start_sfx, end_ip, end_sfx, err = types.Addr2Ints(*decision.Value)
-			if err != nil {
-				return errors.Wrapf(err, "invalid ip/range %s", *decision.Value)
-			}
-		}
+	alertID, err := a.dbClient.CreateSingleAlertWithBulk(CapiMachineID, &capiPullTopX)
+	if err != nil {
+		log.Errorf("error while saving alert from capi/community-blocklist : %s", err)
+	}
 
-		duration, err := time.ParseDuration(*decision.Duration)
-		if err != nil {
-			return errors.Wrapf(err, "parse decision duration '%s':", *decision.Duration)
-		}
-		_, err = a.dbClient.Ent.Decision.Create().
-			SetUntil(time.Now().Add(duration)).
-			SetScenario(*decision.Scenario).
-			SetType(*decision.Type).
-			SetIPSize(int64(sz)).
-			SetStartIP(start_ip).
-			SetStartSuffix(start_sfx).
-			SetEndIP(end_ip).
-			SetEndSuffix(end_sfx).
-			SetValue(*decision.Value).
-			SetScope(*decision.Scope).
-			SetOrigin(*decision.Origin).
-			SetOwner(alertCreated).Save(a.dbClient.CTX)
-		if err != nil {
-			return errors.Wrap(err, "decision creation from crowdsec-api:")
-		}
+	log.Printf("capi/community-blocklist : added %d entries", len(data.New))
+
+	intID, err := strconv.Atoi(alertID)
+	if err != nil {
+		return errors.Wrap(err, "while converting alert ID")
+	}
+	/*Deleting older decisions from capi*/
+	decisions_count, err := a.dbClient.Ent.Decision.Delete().
```

Comment:
Maybe add this in the database pkg as a function to centralize all db operations.